python/tk_katana/__init__.py

```diff
@@ -72,75 +72,3 @@ def __create_tank_error_menu():
     else:
         print("The Shotgun Pipeline Toolkit caught an error: %s" % message)
 
-
-def __engine_refresh(tk, new_context):
-    """
-    Checks the the tank engine should be
-    """
-    engine_name = os.environ.get("TANK_KATANA_ENGINE_INIT_NAME")
-
-    curr_engine = tank.platform.current_engine()
-    if curr_engine:
-        # an old engine is running.
-        if new_context == curr_engine.context:
-            # no need to restart the engine!
-            return
-        else:
-            # shut down the engine
-            curr_engine.destroy()
-
-    # try to create new engine
-    try:
-        tank.platform.start_engine(engine_name, tk, new_context)
-    except tank.TankEngineInitError, e:
-        # context was not sufficient! - disable tank!
-        __create_tank_disabled_menu(e)
-
-
-def __tank_on_scene_event_callback(**kwargs):
-    """
-    Callback that fires every time a file is saved or loaded.
-
-    Carefully manage exceptions here so that a bug in Tank never
-    interrupts the normal workflows in Katana.
-    """
-    # get the new file name
-    file_name = FarmAPI.GetKatanaFileName()
-
-    if not file_name:   # New scene
-        return
-
-    try:
-        # this file could be in another project altogether, so create a new Tank
-        # API instance.
-        try:
-            tk = tank.tank_from_path(file_name)
-        except tank.TankError, e:
-            __create_tank_disabled_menu(e)
-            return
-
-        # try to get current ctx and inherit its values if possible
-        curr_ctx = None
-        if tank.platform.current_engine():
-            curr_ctx = tank.platform.current_engine().context
-
-        # and now extract a new context based on the file
-        new_ctx = tk.context_from_path(file_name, curr_ctx)
-
-        # now restart the engine with the new context
-        __engine_refresh(tk, new_ctx)
-    except Exception, e:
-        __create_tank_error_menu()
-
-
-g_tank_callbacks_registered = False
-
-def tank_ensure_callbacks_registered():
```

Comment:
Instead of removing the code, we could have a setting in the `app.yml` file that would be `callbacks: false` or something like that, and we could check the value of the setting inside this function. If false, we don't create the callback, otherwise, we create them. When developing engines, I think we should always think that we are not the only one in this world (since we forked the engine) and that we might want to share our stuff with Shotgun at some point.